.github/workflows/python-app.yml

```diff
@@ -0,0 +1,37 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python application
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+      - name: Prepare config
+        run: |
+          cp config.py.sample config.py
```

Comment:
this is also ok here.

As a best practice we use tox or our custom paver script that will do this test preparation.

The idea is that you will just run `paver test` ... which if you have not used paver is like `make test` and paver will take care of creating the confing and call the tests.

In this way on both GHA and on you local virtualenv you can just call the exact same command `paver test` ... `python -m paver test` :)

Reply:
I understand. Right now the test instructions are duplicated, in the merged pull request (where nobody will look) and in the GHA. But it is not so difficult to figure it out from the error message either.

As such, I don't think it's worth creating a Paver config for this script. I will leave it as is.

Reply:
Yes. Paver is complex. For simple things you can use `tox`.

* tox - creates virtual env in a controls way and launch simple commands
* paver - launch complex commands with internal dependencies . Doesn't create the virtual env.

In fact, for our main project we have a custom tool that combines paver with our custom `brink` scrip that is used to create a virtualenv.

But I encourage you to check tox as it's very used in open source projects.